File: INTRODUCCION A POO/EJERCICIO-5/Estudiante.java
public class Estudiante {
    private String nombre;
    private double nota1;
    private double nota2;

    public Estudiante(String nombre, double nota1, double nota2) {
        this.nombre = nombre;
        this.nota1 = nota1;
        this.nota2 = nota2;
    }

    public double calcularPromedio() {
        return (this.nota1 + this.nota2) / 2;
    }

    public boolean aprobo() {
        return this.calcularPromedio() >= 6;
    }

    public String getNombre() {
        return this.nombre;
    }

    public static void main(String[] args) {
        Estudiante estudiante1 = new Estudiante("Carlos", 8, 9);
        Estudiante estudiante2 = new Estudiante("Sofía", 4, 7);
        Estudiante estudiante3 = new Estudiante("Pedro", 5, 6);

        System.out.println("Promedio de " + estudiante1.getNombre() + ": " + estudiante1.calcularPromedio() + " - Aprobó: " + estudiante1.aprobo());
        System.out.println("Promedio de " + estudiante2.getNombre() + ": " + estudiante2.calcularPromedio() + " - Aprobó: " + estudiante2.aprobo());
        System.out.println("Promedio de " + estudiante3.getNombre() + ": " + estudiante3.calcularPromedio() + " - Aprobó: " + estudiante3.aprobo());
    }
}

File: INTRODUCCION A POO/EJERCICIO-5/Estudiante.py
class Estudiante:
    def __init__(self, nombre, nota_1, nota_2):
        self.nombre = nombre
        self.nota_1 = nota_1
        self.nota_2 = nota_2
    
    def calcular_promedio(self):
        return (self.nota_1 + self.nota_2) / 2
    
    def aprobo(self):
        return self.calcular_promedio() >= 6

if __name__ == "__main__":
    estudiante1 = Estudiante("Carlos", 8, 9)
    estudiante2 = Estudiante("Sofía", 4, 7)
    estudiante3 = Estudiante("Pedro", 5, 6)
    
    print(f"Promedio de {estudiante1.nombre}: {estudiante1.calcular_promedio()} - Aprobó: {estudiante1.aprobo()}")
    print(f"Promedio de {estudiante2.nombre}: {estudiante2.calcular_promedio()} - Aprobó: {estudiante2.aprobo()}")
    print(f"Promedio de {estudiante3.nombre}: {estudiante3.calcular_promedio()} - Aprobó: {estudiante3.aprobo()}")